File: 0027-remove-element/0027-remove-element.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        rptr, wptr = 0, 0
        while rptr < len(nums):
            if nums[rptr] != val:
                nums[wptr] = nums[rptr]
                wptr += 1
            rptr += 1
        return wptr

File: 0027-remove-element/0027-remove-element.js
/**
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
 */
var removeElement = function(nums, val) {
    let wptr = 0, rptr = 0
    while (rptr < nums.length) {
        if (nums[rptr] === val) rptr+= 1
        else nums[wptr++] = nums[rptr++]
    }
    return wptr
};